core/src/main/scala/org/apache/spark/sql/delta/DeltaSharedExceptions.scala

```diff
@@ -48,3 +58,17 @@ class DeltaUnsupportedOperationException(
     override def getErrorClass: String = errorClass
     def getMessageParametersArray: Array[String] = messageParameters
 }
+
+// todo: we had to add this since in Spark 3.4 ParseException(message, ...) was replaced by
```

Comment:
nit: todo's are generally capitalized.

Reply:
Sorry also didn't plan to merge this todo. Mostly a note to myself or reviewers. I think it's easiest not to add an error class for every invocation in `DeltaSqlParser` but I'm not super clear on how much we want to stick to enforcing using the Spark error framework. I'm happy to do it if we want to stick to the framework